plugins/providers/bigquery/provider_test.go

```diff
@@ -0,0 +1,1110 @@
+		providerURN := "test-URN"
+		crypto := new(mocks.Crypto)
+		client := new(mocks.BigQueryClient)
+		p := bigquery.NewProvider("", crypto)
+		p.Clients = map[string]bigquery.BigQueryClient{
+			"test-resource-name": client,
+		}
+		pc := &domain.ProviderConfig{
+			Resources: []*domain.ResourceConfig{
+				{
+					Type: bigquery.ResourceTypeDataset,
+					Roles: []*domain.Role{
+						{
+							ID:          "VIEWER",
+							Name:        "VIEWER",
+							Permissions: []interface{}{"invalid permissions for resouce type"},
+						},
+					},
+				},
+			},
+			Credentials: bigquery.Credentials{
+				ServiceAccountKey: base64.StdEncoding.EncodeToString([]byte(`{"type":"service_account"}`)), // private_key
+				ResourceName:      "projects/test-resource-name",
+			},
+			URN: providerURN,
+		}
+		//crypto.On("Encrypt", `{"type":"service_account"}`).Return(`{"type":"service_account"}`, nil)
+
+		actualError := p.CreateConfig(pc)
+
+		assert.Error(t, actualError)
+	})
+
+	t.Run("should return error if error in encrypting the credentials", func(t *testing.T) {
+		providerURN := "test-URN"
+		crypto := new(mocks.Crypto)
+		client := new(mocks.BigQueryClient)
+		p := bigquery.NewProvider("", crypto)
+		p.Clients = map[string]bigquery.BigQueryClient{
+			"test-resource-name": client,
+		}
+		pc := &domain.ProviderConfig{
+			Resources: []*domain.ResourceConfig{
+				{
+					Type:  bigquery.ResourceTypeDataset,
+					Roles: []*domain.Role{},
+				},
+			},
+			Credentials: bigquery.Credentials{
+				ServiceAccountKey: base64.StdEncoding.EncodeToString([]byte(`{"type":"service_account"}`)),
+				ResourceName:      "projects/test-resource-name",
+			},
+			URN: providerURN,
+		}
+		expectedError := errors.New("error in encrypting SAK")
+		crypto.On("Encrypt", `{"type":"service_account"}`).Return("", expectedError)
+		actualError := p.CreateConfig(pc)
+
+		assert.Equal(t, expectedError, actualError)
+	})
+
+	t.Run("should return nil error and create the config on success", func(t *testing.T) {
+		providerURN := "test-URN"
+		crypto := new(mocks.Crypto)
+		client := new(mocks.BigQueryClient)
+		p := bigquery.NewProvider("", crypto)
+		p.Clients = map[string]bigquery.BigQueryClient{
+			"test-resource-name": client,
+		}
+		pc := &domain.ProviderConfig{
+			Resources: []*domain.ResourceConfig{
+				{
+					Type: bigquery.ResourceTypeDataset,
+					Roles: []*domain.Role{
+						{
+							ID:          "VIEWER",
+							Name:        "VIEWER",
+							Permissions: []interface{}{"READER"},
+						},
+					},
+				},
+			},
+			Credentials: bigquery.Credentials{
+				ServiceAccountKey: base64.StdEncoding.EncodeToString([]byte(`{"type":"service_account"}`)), // private_key
+				ResourceName:      "projects/test-resource-name",
+			},
+			URN: providerURN,
+		}
+		crypto.On("Encrypt", `{"type":"service_account"}`).Return(`{"type":"service_account"}`, nil)
+
+		actualError := p.CreateConfig(pc)
+
+		assert.NoError(t, actualError)
```

Comment:
let's assert mock expectations at least on the success test cases
```suggestion
		assert.NoError(t, actualError)
		crypto.AssertExpectations(t)
```